neqo-crypto/src/aead.rs

```diff
@@ -126,6 +126,39 @@ impl RealAead {
         Ok(&output[0..(l.try_into()?)])
     }
 
+    /// Encrypt `data` consisting of `aad` and plaintext `input` in place.
+    ///
+    /// The space provided in `data` needs to allow `Aead::expansion` more bytes to be appended.
+    ///
+    /// # Errors
+    ///
+    /// If the input can't be protected or any input is too large for NSS.
+    pub fn encrypt_in_place(
+        &self,
+        count: u64,
+        aad: Range<usize>,
+        input: Range<usize>,
+        data: &mut [u8],
+    ) -> Res<usize> {
+        let aad = &data[aad];
+        let input = &data[input];
```

Comment:
I would be happier if `input` is required to be the plaintext plus `self.expansion()` in length, then you would construct the cipher input as `&mut input[..input.len() - self.expansion()]`.  If you do that, you can drop the `usize` value from the return (success means that you filled the buffer).  The cost of that is that the caller needs to understand how much padding to include, which is pretty-much already the case anyway.